### PR TITLE
snap: show apps in `snap info`

### PR DESCRIPTION
**Pull request by mvo5**
This is a followup from a previous `snap info` branch to display what apps are part of a snap.

There was some disagreement about the format of the displayed apps and concern that the output becomes hard to read. This branch will use the short format when possible and a longer format when the mount of apps is too big for a single line.

Example:
```
$ snap info jenkins
name:      jenkins
summary:   "Jenkins automation server"
publisher: ev
description: |
  Continuous integration and delivery server with a vast ecosystem of plugins
apps:        ["config", "jenkins"]
...
```

And here an artificial example of a snap with many apps (the test-snapd-tools apps fit still into a single line but I tweaked the size to get this output):
```
$ snap info test-snapd-tools
name:      test-snapd-tools
summary:   "Tools for testing the snapd application"
publisher: canonical
description: |
  A tool to test snapd
apps:
  - block
  - cat
  - head
  - env
  - echo
  - success
  - fail
...
`` `

### CODE REVIEW

**Comment by niemeyer** [comment]
Last time we discussed this feature, it wasn't clear what people were actually after. Do we want to show apps, or do we want to show commands (perhaps including aliases later)?  IOW, what are we trying to do by displaying the apps here?

**Comment by mvo5** [comment]
@niemeyer Thanks for your feedback. I reworked the PR to include only the commands for now (and the aliases:
```
$ sudo ./snap info aliases
name:      aliases
summary:   ""
commands:  ["cmd1 (alias1)", "cmd2 (alias2)"]
...
```
and
```
$ snap info hello
name:      hello
commands:  ["universe", "hello"]
...
```
happy to rework this further if the combination of "cmd (alias1,...) is not good, maybe:
```
name: hello
commands: ["cmd1", "cmd2"]
aliases: 
  cmd1: ["alias1"]
  cmd2: ["alias2"]
```
?

File: cmd/snap/cmd_info.go [hunk 1]
```diff
@@ -146,6 +146,42 @@ func formatDescr(descr string, max int) string {
...
+		}
+		commands = append(commands, cmdStr)
+	}
+	if len(commands) == 0 {
+		return
+	}
+
+	// try short line
+	line := fmt.Sprintf("commands:\t[%s]\n", strutil.Quoted(commands))
+	if len(line) < n {
```
Comment (pedronis) [inline cmd/snap/cmd_info.go]: I think the agreement is not to do this and do the long one always

File: cmd/snap/cmd_info.go [hunk 2]
```diff
@@ -146,6 +146,42 @@ func formatDescr(descr string, max int) string {
...
+
+	commands := make([]string, 0, len(allApps))
+	for _, app := range allApps {
+		if app.Daemon != "" {
+			continue
+		}
+
+		var cmdStr string
+		if len(app.Aliases) == 0 {
+			cmdStr = app.Name
```
Comment (pedronis) [inline cmd/snap/cmd_info.go]: if I understood the idea is to use filepath.Base(app.WrapperPath()) instead of app.Name ?

**Pull request merged**